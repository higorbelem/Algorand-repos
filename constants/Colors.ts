const tintColorDark = '#fff';

export const Colors = {
  text: '#FFFFFF',
  background: '#191C21',
  tint: tintColorDark,
  icon: '#9BA1A6',
  tabIconDefault: '#9BA1A6',
  tabIconSelected: tintColorDark,
  main: '#2D2DF1',
  green: '#238636'
};
