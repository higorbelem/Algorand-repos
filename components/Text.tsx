import { StyleSheet, Text as T, TextProps as ReactTextProps } from "react-native"

import { Colors } from "@/constants/Colors";

const sizes = {
    header: 24,
    normal: 16,
    small: 12,
    tiny: 10
}

const weights = {
    bold: "700",
    normal: "400",
}

export type TextProps = ReactTextProps & {
   size?: keyof typeof sizes;
   weight?: keyof typeof weights;
};

export const Text = (props: TextProps) => {
    return(
        <T 
            {...props} 
            style={[
                styles.text, 
                props.style, 
                {
                    fontSize: sizes[props.size || 'normal'],
                    fontWeight: weights[props.weight || 'normal'] as any
                }
            ]}
        >{props.children}</T>
    )
}

const styles = StyleSheet.create({
    text: {
        color: Colors.text
    },
});