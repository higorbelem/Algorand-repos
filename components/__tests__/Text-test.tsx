import * as React from 'react';
import { render } from '@testing-library/react-native';

import { Colors } from '@/constants/Colors';

import { Text } from '../Text';

describe('Text component', () => {
  it('Renders correctly', () => {
    const tree = render(<Text>Snapshot test!</Text>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Shows correct text', async () => {
    const tree = render(<Text>Correct text</Text>);

    expect(await tree.findByText('Correct text')).toBeTruthy();
  });

  it('Has correct color', async () => {
    const tree = render(<Text>Correct text</Text>);
    
    const component = await tree.findByText('Correct text');

    expect(component).toHaveStyle({color: Colors.text});
  });
});