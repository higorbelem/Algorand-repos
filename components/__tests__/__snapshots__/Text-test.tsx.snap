// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component Renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#FFFFFF",
      },
      undefined,
      {
        "fontSize": 16,
        "fontWeight": "400",
      },
    ]
  }
>
  Snapshot test!
</Text>
`;
